src/targets/maven.ts

```diff
@@ -187,31 +187,117 @@ export class MavenTarget extends BaseTarget {
    * @param revision Git commit SHA to be published.
    */
   public async publish(_version: string, revison: string): Promise<void> {
-    await this.createUserGradlePropsFile();
-    await this.upload(revison);
+    await withTempDir(async dir => {
+      const snapshotPath = await this.makeGradlePropsSnapshot(dir);
+      // Overwrite the file, instead of appending the content, because:
+      // 1. The target must get all the config by itself, without relying on
+      // local config files.
+      // 2. Appending more config to the gradle config file may imply a
+      // misconfiguration(such as duplicated or conflicting keys).
+      await this.createUserGradlePropsFile();
+
+      await this.upload(revison);
+
+      // Maven central is very flaky, so retrying with an exponential delay in
+      // in case it fails.
+      // TODO: close the repository by doing the requests from Craft
+      // What the gradle plugin does is a / some requests to close the repo, see
+      // https://github.com/vanniktech/gradle-maven-publish-plugin/tree/master/src/main/kotlin/com/vanniktech/maven/publish/nexus
+      // If Craft did those requests, it wouldn't be necessary to rely on a third
+      // party plugin for releases.
+      await retrySpawnProcess(this.mavenConfig.gradleCliPath, [
+        'closeAndReleaseRepository',
+      ]);
+      await this.recoverGradlePropsSnapshot(snapshotPath);
+    });
+  }
 
-    // Maven central is very flaky, so retrying with an exponential delay in
-    // in case it fails.
-    // TODO: close the repository by doing the requests from Craft
-    // What the gradle plugin does is a / some requests to close the repo, see
-    // https://github.com/vanniktech/gradle-maven-publish-plugin/tree/master/src/main/kotlin/com/vanniktech/maven/publish/nexus
-    // If Craft did those requests, it wouldn't be necessary to rely on a third
-    // party plugin for releases.
-    await retrySpawnProcess(this.mavenConfig.gradleCliPath, [
-      'closeAndReleaseRepository',
-    ]);
-    await this.deleteUserGradlePropsFile();
+  /**
+   * Makes a snapshot of the gradle properties file, if exists.
+   *
+   * @param snapshotDir Directory to place the snapshot in.
+   * @returns The path to the snapshot in the given directory,
+   *  or `undefined` if the snapshot wasn't created.
+   */
+  public async makeGradlePropsSnapshot(
+    snapshotDir: string
+  ): Promise<string | undefined> {
+    const propsExpectedPath = this.getGradlePropsPath();
+    try {
+      await fsPromises.access(propsExpectedPath, fsConstants.R_OK);
```

Comment:
You don't need this step. Just move this `try`-`catch` case down to the `copyFile` call and it is fine. This way you'll save an IO op _and_ a thread cycle.